core/kernel/classes/class.Resource.php

```diff
@@ -393,6 +395,8 @@ public function setPropertyValue( core_kernel_classes_Property $property, $objec
     {
         $returnValue = (bool) false;
         $returnValue = $this->getImplementation()->setPropertyValue($this, $property, $object);
+        $eventManager = $this->getServiceManager()->get(EventManager::SERVICE_ID);
```

Comment:
Pleas create `\core_kernel_classes_Resource::onUpdate()` function and call it to trigger event. It will reduce amount of lines of code and make triggering this event centralized.